docs/design/authorization.md

```diff
@@ -0,0 +1,21 @@
+# Authorization
+
+## Problem statement
+
+Smith typically has a very broad set of permissions because it needs to be able to manage objects of various kinds
+across several/many namespaces in a cluster. If permissions checks are not done properly, a malicious user may have
+a way to escalate their privileges.
+This document describes a way to address the problem of blocking ways to permissions escalation.
+
+## Proposed solution
+
+The idea is to capture identity of the user who creates/updates a `Bundle` and impersonate them when working with
+objects of that `Bundle`. That way Smith becomes just a tool that automates operations of behalf of the user rather
+than a more privileged actor that can do something the user cannot.
```

Comment:
Being able to impersonate **any** user still makes Smith a privileged actor.
It just dramatically reduces the number of ways of abusing it.
Not sure how to better phrase this though.